src/main/java/org/humancellatlas/ingest/security/WhitelistJwtAuthenticationProvider.java

```diff
@@ -0,0 +1,41 @@
+package org.humancellatlas.ingest.security;
+
+import com.auth0.spring.security.api.JwtAuthenticationProvider;
+import org.humancellatlas.ingest.security.exception.UnlistedEmail;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+
+import javax.annotation.Nullable;
+
+import static java.util.Optional.ofNullable;
+
+public class WhitelistJwtAuthenticationProvider implements AuthenticationProvider {
+
+    private final JwtAuthenticationProvider delegate;
+
+    private final UserWhiteList userWhitelist;
+
+    public WhitelistJwtAuthenticationProvider(JwtAuthenticationProvider delegate, UserWhiteList userWhitelist) {
+        this.delegate = delegate;
+        this.userWhitelist = userWhitelist;
+    }
+
+    @Override
+    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+        Authentication jwtAuthentication = delegate.authenticate(authentication);
+        ofNullable(jwtAuthentication).ifPresent(auth -> {
+            String principal = auth.getPrincipal().toString();
```

Comment:
which attribute in the jwt token payload is the principal? is it the user email? 
```{
  "https://auth.data.humancellatlas.org/email": "aaclan@ebi.ac.uk",
  "https://auth.data.humancellatlas.org/group": "hca",
  "iss": "https://humancellatlas.auth0.com/",
  "sub": "google-oauth2|113069531154039191158",
  "aud": [
    "https://dev.data.humancellatlas.org/",
    "https://humancellatlas.auth0.com/userinfo"
  ],
  "iat": 1574096428,
  "exp": 1574182828,
  "azp": "podw-relvCupzkJctqjFO0y5RQQoiUF6",
  "scope": "openid profile email"
}
```

Instead of checking the actual user email, might be better to just check that the value of  `https://auth.data.humancellatlas.org/group`  attribute in the token is `hca` and not `public`. There is a rule setup in HCA Auth0 to set this group attr if it has valid email domain (ebi, czi, broad, stanford etc).